nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/internal/HelmHostedIT.java

```diff
@@ -0,0 +1,169 @@
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.helm.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.core.MediaType;
+
+import org.sonatype.nexus.common.app.BaseUrlHolder;
+import org.sonatype.nexus.pax.exam.NexusPaxExamSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.ComponentMaintenance;
+import org.sonatype.nexus.testsuite.testsupport.NexusITSupport;
+
+import org.apache.http.HttpResponse;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+
+import static java.lang.String.format;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class HelmHostedIT
+    extends HelmITSupport
+{
+  private HelmClient client;
+
+  private Repository repository;
+
+  @Configuration
+  public static Option[] configureNexus() {
+    return NexusPaxExamSupport.options(
+        NexusITSupport.configureNexusBase(),
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-helm")
+    );
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    BaseUrlHolder.set(this.nexusUrl.toString());
+    repository = repos.createHelmHosted("helm-hosted-test");
+    client = createHelmClient(repository);
+  }
+
+  @Test
+  public void testPackageUpload() throws IOException
+  {
+    uploadPackages(MONGO_PKG_FILE_NAME_600_TGZ, MONGO_PKG_FILE_NAME_728_TGZ);
+    //Verify DB contains data about uploaded component and asset
+    Component component = findComponent(repository, MONGO_PKG_NAME);
+    assertThat(component.name(), is(equalTo(MONGO_PKG_NAME)));
+    assertThat(component.version(), is(equalTo(MONGO_PKG_VERSION_600)));
+
+    //Verify Asset is created.
+    Asset asset = findAsset(repository, MONGO_PATH_FULL_600_TARGZ);
+    assertThat(asset.name(), is(equalTo(MONGO_PATH_FULL_600_TARGZ)));
+    assertThat(asset.format(), is(equalTo(HELM_FORMAT_NAME)));
+  }
+
+  @Test
+  public void testFetchPackageOnEmptyRepository() throws Exception
+  {
+    HttpResponse resp = client.fetch(MONGO_PATH_FULL_728_TARGZ, CONTENT_TYPE_TGZ);
+    assertThat(resp.getEntity().getContentType().getValue(), equalTo(MediaType.TEXT_HTML));
+  }
```

Comment:
@Darkvar Thanks. Would you mind adding a verification of the return code for that case? e.g. something like:

```
    MatcherAssert.assertThat(status(resp), is(HttpStatus.NOT_FOUND));
```